### PR TITLE
[READY]Clownborgs - ADMIN SPAWN ONLY

### PR DESCRIPTION
**Pull request by silicons**
image
Sprites by the amazing @keekenox !

:cl:
rscadd: Clown modules have been added for cyborgs! Modules: Bikehorn, Airhorn, instrumental bikehorn, clown stamp, multicolor paint, rainbow crayon, soap, razor, purple lipstick, nonstunning selfcharging creampie cannon, nonslipping selfcharging water sprayer, lollipop fabricator, metallic picket sign, mini extinguisher, laughter reagent hypospray, and the ability to hug.
experimental: When emagged they get a fire spray, "super" laughter injector, and shocking or crushing hugs. Also, the module is adminspawn only at the moment.
/:cl:
Modules: [I'm going to mark the ones useful in combat in any shape or form.]
flash [WEAPON - MELEE - Stun]
rainbow crayon + multicolor paint + clown stamp
bikehorn + airhorn + bikehorn instrument
razor + purple lipstick
nanotrasen soap
non-stunning creampie cannon [WEAPON - MELEE - 10 force brute]
non-slipping water-only waterflower
lollipop fabricator 
metallic picket sign [WEAPON - MELEE - 13 force brute]
mini extinguisher [WEAPON - MELEE - something-that-is-less-than-10 brute]
laughter reagent hypospray
cyborg hug

Emagged modules/upgrades:
laughter reagent hypospray that stuns on a 30% chance/cycle, same metabolism rate as chloral (higher than normal)
clf3 water flower with 2 charges and slow charge rate
Cyborg hug gains melee stun and damage like peacekeeper hug module.

Ratvar modules:
Same as standard borg for now.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
I think you mistakenly got the wrong date. This is December, not April.

**Comment by kevinz000** [comment]
Who said I was joking?
🤡

**Comment by Bawhoppen** [comment]
The 13-brute picket sign is what gave it away

**Comment by kevinz000** [comment]
eh i was going to use it because i didn't want to have an extinguisher but whatever.

**Comment by Tlaltecuhtli** [comment]
can you make clown borg module upgrades that require banarium that unlocks something like the clown shoes prototype

**Comment by ShiggyDiggyDo** [comment]
Borgs are there to serve humans, not to prank them

**Comment by MoreRobustThanYou** [comment]
@ShiggyDiggyDo Then why do peacekeepers exist? They literally only dispense cookies, **_HARM ALARM_**, and hug people.

**Comment by ShiggyDiggyDo** [comment]
Peacekeepers are supposed to break up fights and prevent human harm, but most people just use them as memeborgs when they get forceborged

**Comment by kevinz000** [comment]
sir the correct question is not "why" but "why not".

**Comment by arsserpentarium** [comment]
That is hoid work!
This is what you should do instead of nerfing!

**Comment by praisenarsie** [comment]
@Tlaltecuhtli you are very attention-seeking

**Comment by The-clown-known-as-Pepe** [comment]
YOU'RE USING THE WRONG SPRITE 
image

**Comment by Xhuis** [comment]
It's amusing, but I can't really think of any functional reason to have this. I probably sound like a jaded antifun jerk, meh. I just don't really get the point!

**Comment by The-clown-known-as-Pepe** [comment]
@Xhuis literally the clown role in a nutshell

**Comment by GunHog** [comment]
The modules are fine, but the original sprite proposal is lacking. Considering the nature of this borg being mostly entertainment and annoyance, consider giving it skin choices, like the service borg does. Starting with the Peacekeeper version given by @The-clown-known-as-Pepe would be a good start.

**Comment by praisenarsie** [comment]
if we are giving it the peacekeeper sprite can we remove peacekeepers then

**Comment by The-clown-known-as-Pepe** [comment]
@GunHog i asked my friend to make this sprite specificaly for this PR

**Comment by Xhuis** [comment]
@The-clown-known-as-Pepe Everyone knows the clown exists to be shot, stabbed, beaten, set on fire, etc.! Borgs are immune to two damage types and killing them is more likely to get you bwoinked!

**Comment by The-clown-known-as-Pepe** [comment]
Or you can just order them to leave you alone.

**Comment by GunHog** [comment]
If you do not want to be beaten, do not play as a punching bag!

**Comment by Xhuis** [comment]
*punches @GunHog*

**Comment by ShizCalev** [comment]
I fear that God's a ship
And he's thrown us into a great pot
He is here, fully formed
Squeaking, red, and iron-wrought

**Comment by pyronie** [comment]
make them be able to spawn soap for 5 thousand power or so.
let the station feel havoc whenever a clown borg appears

**Comment by newfren** [comment]
a clown borg being unable to slip is a great injustice

**Comment by coiax** [comment]
Why not just add all of this stuff to PK borgs and give them an alternate skin?

**Comment by The-clown-known-as-Pepe** [comment]
@coiax that would make pk borgs broken

**Comment by coiax** [comment]
Broken as too powerful?

**Comment by The-clown-known-as-Pepe** [comment]
yeah i feel like stuffing 20 modules on one borg is a little too much

**Comment by coiax** [comment]
I mean, it looks like a mix between PK and Service borg, frankly.

**Comment by ShizCalev** [comment]
pepe's sprite or bust

**Comment by The-clown-known-as-Pepe** [comment]
sprite was made by keekenox

**Comment by Doppeltongue** [comment]
What if you upgraded PK borgs into clownborgs by shoving in a clown's wig/mask into an open PK borg?

**Comment by Anonymous583** [comment]
Keep shitkeepers away from pure clownborgs.

**Comment by RemieRichards** [comment]
One day all jobs will be PK Borgs.

Imgur

It is inevitable.

**Comment by GunHog** [comment]
Those NEED to exist, now...

**Comment by RemieRichards** [comment]
@GunHog  I made them to Xhuis for april fools but he never used them.
https://i.imgur.com/cjpd3cY.png - Be my guest (convert from .png -> .dmi, the metadata is still there last I checked)

**Comment by Xhuis** [comment]
Yeah, I couldn't ever figure it out. I *still* can't get the appearances system to work whenever I use it, but I guess that's just inexperience with it on my part.

I might try it again for 2018 April Fools', dunno though.

**Comment by GunHog** [comment]
These should be skin choices all-around

**Comment by BeeSting12** [comment]
make the pie cannon stun

**Comment by epochayur** [comment]
fucking terrible, why on earth would a silicon exist to just annoy people?

**Comment by The-clown-known-as-Pepe** [comment]
because its in spirit of the game

**Comment by epochayur** [comment]
Having a damage-resistant clownborg flashing me in the hallway, shaving me bald and injecting me with laughing chems, then sprinting off is not my idea of a good time. And, again, why is this allowable default behavior for a silicon, supposedly a being whose soul purpose is to aid humans, not to annoy the shit out of them?

**Comment by The-clown-known-as-Pepe** [comment]
@epochayur "Having a damage-resistant clownborg flashing me in the hallway, shaving me bald and injecting me with laughing chems, then sprinting off is not my idea of a good time."

Or you could order the borg to stop if you weren't focusing your braincells to strawman this hard.

**Comment by epochayur** [comment]
or just, you know, never merge it, considering we already have two jobs that are dedicated to fucking with people

**Comment by The-clown-known-as-Pepe** [comment]
Or you could order the borg to pick another module if you weren't focusing your braincells to strawman this hard.

**Comment by epochayur** [comment]
then why have it to begin with? @kevinz000 call off your apologists they're hurting my feefees

**Comment by The-clown-known-as-Pepe** [comment]
make the game fun for me and me only t. epochayur

**Comment by epochayur** [comment]
untitled

yeah it's definitely only me

**Comment by The-clown-known-as-Pepe** [comment]
image

Everyone is reacting with up thumb emoji here meaning you're wrong

Also come up with something that's not a strawman pls

**Comment by epochayur** [comment]
learn what a strawman is

**Comment by The-clown-known-as-Pepe** [comment]
Please stay on the topic 🤗

**Comment by PKPenguin321** [comment]
Terrible sprite kev

**Comment by optimumtact** [comment]
I believe that on the balance this should be allowed, however there needs to be some gate, this can't just be a module any borg can pick.

My initial thought was simply to make the module only appear when a techweb entry was reached, but I thought it would be more thematic if there was an organ called the clown's heart (inside every joker is one), that lives inside every clown and you have to cut the heart out of your existing clown to make a clown borg or get a clown heart via some other means.

With high enough techweb research you could make a clown heart out of bananium and not have to murder your existing clown to make a clown borg.


I know this is extra code, but you can add the module as an admin spawn only item for now and work on that in a separate PR if you please.

**Comment by kevinz000** [comment]
ok @optimumtact 
@ShizCalev im adding an option for it soonish

**Comment by kevinz000** [comment]
they're now using the new sprites by @keekenox.

**Comment by The-clown-known-as-Pepe** [comment]
>you can revive dead clowns as clownborgs

great idea citrus

**Comment by imsxz** [comment]
if it's admin spawn only, I don't see the issue with giving it banana peel synthesizers and stunning pie cannon?

**Comment by optimumtact** [comment]
I think you didn't read my comment all the way through

**Comment by The-clown-known-as-Pepe** [comment]
here's a better idea than what citrus proposed

make them be enabled as a choice by an upgrade that requires clown tech and bananium

**Comment by The-clown-known-as-Pepe** [comment]
>adminspawn only

🤢

**Comment by Keekenox** [comment]
Admin spawn only? That makes me sad.

**Comment by Anonymous583** [comment]
Admin spawn only is stupid. Make it roundstart.

**Comment by optimumtact** [comment]
thank you for your opinions

File: code/game/objects/items/signs.dm [hunk 1]
```diff
@@ -10,13 +10,23 @@
...
+	desc = "A high tech picket sign used by silicons that can reprogram its surface at will. Probably hurts to get hit by, too."
+	force = 13
+	resistance_flags = NONE
+	actions_types = list(/datum/action/item_action/nano_picket_sign)
+
+/obj/item/picket_sign/proc/retext(mob/user)
+	var/txt = stripped_input(user, "What would you like to write on the sign?", "Sign Label", null , 30)
+	if(txt && Adjacent(user))
+		label = txt
+		src.name = "[label] sign"
```
Comment (optimumtact) [inline code/game/objects/items/signs.dm]: Boy I about to suplex you from the top rope

File: code/modules/reagents/reagent_containers/spray.dm [hunk 2]
```diff
@@ -204,6 +204,45 @@
...
+/obj/item/reagent_containers/spray/waterflower/cyborg/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/reagent_containers/spray/waterflower/cyborg/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/reagent_containers/spray/waterflower/cyborg/process()
+	if(++generate_tick < generate_ticks)
```
Comment (optimumtact) [inline code/modules/reagents/reagent_containers/spray.dm]: change this to a world timer

**Comment by arsserpentarium** [comment]
No!admin spawn only is killing all the fun! There is no point in this PR this way!

**Pull request merged**